custom_components/zwave_mqtt/__init__.py

```diff
@@ -246,6 +250,35 @@ async def handle_remove_node(hass: HomeAssistant, node_id: int):
             dev_registry.async_remove_device(dev_id)
 
 
+async def handle_node_update(hass: HomeAssistant, node: OZWNode):
+    """
+    Handle a node updated event from OZW.
+
+    Meaning some of the basic info like name/model is updated.
+    We want these changes to be pushed to the device registry.
+    """
+    dev_registry = await get_dev_reg(hass)
+    # grab device in device registry attached to this node
+    device = dev_registry.async_get_device([(DOMAIN, node.id)], [])
+    if device:
+        # update device in device registry with (updated) info
+        # also handle any slave devices (node instances)
+        for item in dev_registry.devices.values():
+            if item.id == device.id or item.via_device_id == device.id:
+                dev_name = f"{node.node_manufacturer_name} {node.node_product_name}"
+                if item.via_device_id:
+                    instance = list(item.identifiers)[0][2]
```

Comment:
You cannot expect this to be in any order. You need to lookup based on the key.

Reply:
You're right, I took the shortcut. Will adjust